crates/sqlite-inproc/src/lib.rs

```diff
@@ -31,77 +33,99 @@ impl InProcConnection {
     }
 }
 
+#[async_trait]
 impl Connection for InProcConnection {
-    fn query(
+    async fn query(
         &self,
         query: &str,
-        parameters: Vec<spin_world::sqlite::Value>,
-    ) -> Result<spin_world::sqlite::QueryResult, spin_world::sqlite::Error> {
-        let conn = self.connection.lock().unwrap();
-        let mut statement = conn
-            .prepare_cached(query)
-            .map_err(|e| spin_world::sqlite::Error::Io(e.to_string()))?;
-        let columns = statement
-            .column_names()
-            .into_iter()
-            .map(ToOwned::to_owned)
-            .collect();
-        let rows = statement
-            .query_map(
-                rusqlite::params_from_iter(convert_data(parameters.into_iter())),
-                |row| {
-                    let mut values = vec![];
-                    for column in 0.. {
-                        let value = row.get::<usize, ValueWrapper>(column);
-                        if let Err(rusqlite::Error::InvalidColumnIndex(_)) = value {
-                            break;
-                        }
-                        let value = value?.0;
-                        values.push(value);
-                    }
-                    Ok(spin_world::sqlite::RowResult { values })
-                },
-            )
-            .map_err(|e| spin_world::sqlite::Error::Io(e.to_string()))?;
-        let rows = rows
-            .into_iter()
-            .map(|r| r.map_err(|e| spin_world::sqlite::Error::Io(e.to_string())))
-            .collect::<Result<_, spin_world::sqlite::Error>>()?;
-        Ok(spin_world::sqlite::QueryResult { columns, rows })
+        parameters: Vec<sqlite::Value>,
+    ) -> Result<sqlite::QueryResult, sqlite::Error> {
+        let connection = self.connection.clone();
+        let query = query.to_owned();
+        // Tell the tokio runtime that we're going to block while making the query
+        tokio::task::spawn_blocking(move || make_query(&connection, &query, parameters))
+            .await
+            .context("internal runtime error")
+            .map_err(|e| sqlite::Error::Io(e.to_string()))?
     }
 
-    fn execute_batch(&self, statements: &str) -> anyhow::Result<()> {
-        let conn = self.connection.lock().unwrap();
-        conn.execute_batch(statements)?;
+    async fn execute_batch(&self, statements: &str) -> anyhow::Result<()> {
+        let connection = self.connection.clone();
+        let statements = statements.to_owned();
+        tokio::task::spawn_blocking(move || {
+            let conn = connection.lock().unwrap();
+            conn.execute_batch(&statements)
+                .context("failed to execute batch statements")
+        })
+        .await?
+        .context("failed to spawn blocking task")?;
         Ok(())
     }
 }
 
+fn make_query(
```

Comment:
"make" reads ambiguously in this context - you intend it in the sense of "the client makes a call to the database" but I first read it was "create a query object".  May I suggest something like `execute_query` or `perform_query`?

Reply:
Renamed to `execute_query`